### DISCUSSION TITLE
Jekyll 4.0, GitHub Pages and Actions

### DISCUSSION THREAD

**Discussion by oDinZu** [community/community#26041] [Other Feature Feedback, Questions, & Ideas]
<p>I setup GitHub Actions, but have had no luck getting the menus to populate. Would you have a working example with gh-pages branch for Jekyll website and this plug-in?</p>
<p>I tested 9 different configuration setups and searched through the docs, plus if anyone else had any issues like this with Jekyll 4.0. I found this: <a href="https://github.com/helaili/jekyll-action/issues/45" rel="nofollow noopener">https://github.com/helaili/jekyll-action/issues/45</a> and is closest issue I have had similar to mine.</p>
<p>I was unable to successfully build with Jekyll GitHub Actions 2.0.3 or 2.0.2; 2.0.1 enabled me to compile successfully, but even after success, it didn’t merge repo into the gh-pages branch as the documentation states; plus the Jekyll site with this plugin didn’t populate the menu.</p>
<p>When I push changes, I push them to master repository, then GH actions checks, then is suppose to push to gh-pages to allow more possibility with Jekyll plugins and other features.</p>
<p>Below, I have added some code and also a link to my repository.<br>
Repository: <a href="https://github.com/Galactic-Society/galactic-society.github.io/" rel="nofollow noopener">https://github.com/Galactic-Society/galactic-society.github.io/</a> (the GH-Pages setting is set to master @ /root) Also, the repository is an organization and is public.</p>
<p><strong>github-pages.yml</strong></p>
<pre><code class="lang-auto">name: Testing the GitHub Pages publication

on:
  push

jobs:
  jekyll:
    runs-on: ubuntu-16.04
    steps:
    - uses: actions/checkout@v2

    # Use GitHub Actions' cache to shorten build times and decrease load on servers
    - uses: actions/cache@v1
      with:
        path: vendor/bundle
        key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
        restore-keys: |
          ${{ runner.os }}-gems-

    # Standard usage
    - uses:  helaili/jekyll-action@2.0.1
      env:
        JEKYLL_PAT: ${{ secrets.JEKYLL_PAT }}

    # Specify the target branch (optional)
    - uses: helaili/jekyll-action@2.0.1
      env:
        JEKYLL_PAT: ${{ secrets.JEKYLL_PAT }}
      with:
        jekyll_src: 'gh-pages'
</code></pre>
<p><strong>logs</strong></p>
<pre><code class="lang-auto">...
      Generating... 
       Jekyll Feed: Generating feed for posts
DEPRECATION WARNING on line 7 of /github/workspace/_scss/libs/_breakpoints.scss:
!global assignments won't be able to declare new variables in future versions.
Consider adding `$breakpoints: null` at the top level.

                    done in 0.92 seconds.
 Auto-regeneration: disabled. Use --watch to enable.
Jekyll build done
Cannot publish on branch master
</code></pre>

**Comment by oDinZu** [comment]
<p>When using this code: (Only thing changed was the version numbers to 2.0.1 and v1 for cache.</p>
<pre><code class="lang-auto">name: Build and deploy Jekyll site to GitHub Pages

on:
  push:
    branches:
      - master

jobs:
  github-pages:
    runs-on: ubuntu-16.04
    steps:
      - uses: actions/checkout@v1
      - uses: helaili/jekyll-action@2.0.1
        env:
          JEKYLL_PAT: ${{ secrets.JEKYLL_PAT }}
</code></pre>
<p>the build is successful, but does not push or create to <strong>gh-pages</strong> and populates this error:</p>
<pre><code class="lang-auto">
error: src refspec ***@github.com/Galactic-Society/galactic-society.github.io.git does not match any

error: failed to push some refs to 'https://'
</code></pre>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/csharpee">@csharpee</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>I forked your repository, and tried with version 2.0.1, it’s successful to create gh-pages branch and force push to it.</p>
<p>My workflow: <a href="https://github.com/WadeOrg/galactic-society.github.io/runs/1040605985?check_suite_focus=true#step:5:134" rel="nofollow noopener">https://github.com/WadeOrg/galactic-society.github.io/runs/1040605985?check_suite_focus=true#step:5:134</a></p>
<p>My GH page setting:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/eb73c293bdde5bc04a29b768f2528175cfdcaf06.png" data-download-href="/uploads/short-url/xAUdF2KE5tQ8qwYnnkWWhZoXiqG.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181094931-968a3d9b-b834-4dff-a272-91eaee236218.png" alt="image" data-base62-sha1="xAUdF2KE5tQ8qwYnnkWWhZoXiqG" width="657" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/eb73c293bdde5bc04a29b768f2528175cfdcaf06_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">747×568 26.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Could you please check again?</p>
<p>Thanks</p>

**Comment by oDinZu** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a>, thank you for testing this.</p>
<p>I created a commit on GitHub to test again, but no success on a new GH-Pages branch.<br>
<code> error: src refspec ***@github.com/Galactic-Society/galactic-society.github.io.git does not match any</code></p>
<p>My organizational settings are the same as yours.</p>
<p>I don’t know how it works for you, but not me; maybe its a permission thing? I setup my JEKYLL_PAT via <a href="https://github.com/settings/tokens" rel="nofollow noopener">https://github.com/settings/tokens</a> with permission of <strong>public_repo</strong> as the documentation states.</p>
<p>Then, went to organizational account &gt; settings &gt; secrets and copied key from above to JEKYLL_PAT.</p>
<p>Now, going through the steps again; I didn’t use an “organization secret”, but a “repository secret”.<br>
After testing this, it didn’t work either and produced same error shared above.</p>

**Comment by oDinZu** [comment]
<p>Furthermore, I forked this repository on another organizational account and only added the token and updated the settings to enable GH-Pages on master.</p>
<p>After this, the repository workflow created the gh-pages branch on the new organizational account. Maybe the dash - between Galactic-Society caused an issue? This is a very weird bug.</p>
<p>Proof the setup works; only thing different is org account name.<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/SharpeTronics/galactic-society.github.io/runs/1042853376?check_suite_focus=true" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/18726788/181094933-6f4294c7-a4cb-465f-b080-acf11b195d2b.png" class="thumbnail onebox-avatar" width="420" height="420">

<h3><a href="https://github.com/SharpeTronics/galactic-society.github.io/runs/1042853376?check_suite_focus=true" target="_blank" rel="nofollow noopener">SharpeTronics/galactic-society.github.io</a></h3>

<p>Will update description at a later date. Contribute to SharpeTronics/galactic-society.github.io development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by oDinZu** [comment, marked as answer]
<p>For others who may have difficulties with this, I will share a successful result here.</p>
<aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/jekyll/jekyll/issues/8360#issuecomment-683134170" target="_blank" rel="nofollow noopener">github.com/jekyll/jekyll</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/jekyll/jekyll/issues/8360#issuecomment-683134170" target="_blank" rel="nofollow noopener">docs: GitHub-Actions.md</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2020-08-27" data-time="18:51:02" data-timezone="UTC">06:51PM - 27 Aug 20 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/csharpee" target="_blank" rel="nofollow noopener">
          <img alt="csharpee" src="https://user-images.githubusercontent.com/18726788/181094934-7481e2c7-2d56-4568-9c03-643295ce96b1.jpeg" class="onebox-avatar-inline" width="20" height="20">
          csharpee
        </a>
      </div>
    </div>
  </div>
</div>

<div class="github-row">
  <p class="github-content">The page referring to is here: https://jekyllrb.com/docs/continuous-integration/github-actions/
Motivation
After following instructions on the github-actions.md page, it seems the instructions are clear, but do...</p>
</div>

<div class="labels">
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">documentation</span>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
